File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/converters/ConverterListString.java
package pl.szczodrzynski.edziennik.data.db.converters;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.util.List;

import androidx.room.TypeConverter;

public class ConverterListString {

    @TypeConverter
    public static List<String> toListString(String value) {
        return value == null ? null : new Gson().fromJson(value, new TypeToken<List<String>>(){}.getType());
    }

    @TypeConverter
    public static String toString(List<String> value) {
        return value == null ? null : new Gson().toJson(value);
    }
}

File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/converter/ConverterListString.kt
package pl.szczodrzynski.edziennik.data.db.converter

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

class ConverterListString {
    @TypeConverter
    fun toListString(value: String?): List<String>? = value?.let { Gson().fromJson<List<String>>(it, object : TypeToken<List<String?>?>() {}.type) }

    @TypeConverter
    fun toString(value: List<String?>?): String? = value?.let { Gson().toJson(it) }
}
